File: parity/corpus/javascript/report.js
// A branchy classifier plus a risky dynamic eval.
class Report {
  constructor(name, threshold = 70) {
    this.name = name;
    this.threshold = threshold;
  }

  classify(score, flags) {
    let grade;
    if (score > 90) grade = "A";
    else if (score > 80) grade = "B";
    else if (score > this.threshold) grade = "C";
    else grade = "F";
    if (flags && (flags.includes("urgent") || flags.includes("review"))) grade += "!";
    for (const extra of flags) {
      if (extra.startsWith("bonus") && grade !== "F") {
        grade = "A+";
        break;
      }
    }
    return grade;
  }
}

function run(expr) {
  return eval(expr);
}

File: parity/corpus/python/report.py
"""A branchy classifier plus a deliberately risky shell call."""

import os


class Report:
    def __init__(self, name: str, threshold: int = 70) -> None:
        self.name = name
        self.threshold = threshold

    def classify(self, score: int, flags: list[str]) -> str:
        if score > 90:
            grade = "A"
        elif score > 80:
            grade = "B"
        elif score > self.threshold:
            grade = "C"
        else:
            grade = "F"
        if flags and ("urgent" in flags or "review" in flags):
            grade += "!"
        for extra in flags:
            if extra.startswith("bonus") and grade != "F":
                grade = "A+"
                break
        return grade


def run(cmd: str) -> int:
    # dangerous call: exercises the SECURE generator
    return os.system(cmd)
